src/UserService.Business/Commands/Avatar/AddAvatarsCommand.cs

```diff
@@ -0,0 +1,150 @@
+
+namespace LT.DigitalOffice.UserService.Business.Commands.Avatar
+{
+  public class AddAvatarsCommand : IAddAvatarsCommand
+  {
+    private readonly IAvatarRepository _avatarRepository;
+    private readonly IUserRepository _userRepository;
+    private readonly IAccessValidator _accessValidator;
+    private readonly IAddAvatarRequestValidator _requestValidator;
+    private readonly IDbUserAvatarMapper _dbUserAvatarMapper;
+    private readonly ICreateImageDataMapper _createImageDataMapper;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IRequestClient<ICreateImagesRequest> _rcCreateImage;
+    private readonly ILogger<AddAvatarsCommand> _logger;
+
+    public AddAvatarsCommand(
+      IAvatarRepository avatarRepository,
+      IUserRepository userRepository,
+      IAccessValidator accessValidator,
+      IAddAvatarRequestValidator requestValidator,
+      IDbUserAvatarMapper dbUserAvatarMapper,
+      ICreateImageDataMapper createImageDataMapper,
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<ICreateImagesRequest> rcCreateImage,
+      ILogger<AddAvatarsCommand> logger)
+    {
+      _avatarRepository = avatarRepository;
+      _userRepository = userRepository;
+      _accessValidator = accessValidator;
+      _requestValidator = requestValidator;
+      _dbUserAvatarMapper = dbUserAvatarMapper;
+      _createImageDataMapper = createImageDataMapper;
+      _httpContextAccessor = httpContextAccessor;
+      _rcCreateImage = rcCreateImage;
+      _logger = logger;
+    }
+
+    private List<Guid> AddImages(List<AddImageRequest> request, Guid userId, Guid senderId, List<string> errors)
+    {
+      if (request == null || request.Contains(null))
+      {
+        return null;
+      }
+
+      List<Guid> result = null;
+
+      string errorMessage = "Can not add images. Please try again later.";
+      string logMessage = "Errors while adding images.";
+
+      try
+      {
+        Response<IOperationResult<ICreateImagesResponse>> createResponse = _rcCreateImage.GetResponse<IOperationResult<ICreateImagesResponse>>(
+          ICreateImagesRequest.CreateObj(
+            _createImageDataMapper.Map(request, senderId),
+            ImageSource.User))
+          .Result;
+
+        if (createResponse.Message.IsSuccess)
+        {
+          result = createResponse.Message.Body.ImagesIds;
+
+          List<DbUserAvatar> dbUsersAvatarsCreate = _dbUserAvatarMapper.Map(result, userId);
+
+          _avatarRepository.Create(dbUsersAvatarsCreate);
+        }
+        else
+        {
+          string warningMessage = logMessage + " Errors: {Errors}";
+
+          _logger.LogWarning(
+            warningMessage,
+            string.Join('\n', createResponse.Message.Errors));
+
+          errors.Add(errorMessage);
+        }
+      }
+      catch (Exception e)
+      {
+        _logger.LogError(e, logMessage);
+
+        errors.Add(errorMessage);
+      }
+
+      return result;
```

Comment:
May be remove 71 line and return null?

Reply:
maybe
i tried not to use many outs from method